docs/blocks/states.md

```diff
@@ -1,82 +1,98 @@
+An example is making chair blocks: the *direction* of the chair should be a *property*, while the different *types of wood* should be separated into different blocks.
+An "Oak Chair" facing east (`oak_chair[facing=east]`) is different from a "Spruce Chair" facing west (`spruce_chair[facing=west]`).
 
-Every block will also have a "default" state that is automatically chosen for you. You can overwrite this "default" by overwriting the `getDefaultState()` method. More importantly, when your block is placed it will become this "default" state. However if you wish to customise which `BlockState` is placed when you your block is placed you can overwrite `getStateForPlacement()`. This can be used to for example set the direction of your block depending on where the player is standing when they place it.
+Implementing Block States
+---------------------------------------
 
-`BlockState`'s are immutable and pregenerated, this means calling `BlockState.with(<PROPERTY>, <NEW_VALUE>)` will simply go to the `BlockState`/`StateContainer` and request the BlockState with the set of values you want, instead of constructing a new `BlockState`.
+In your Block class, create `static final` `IProperty<?>` objects for every property that your Block has. You are free to make your own `IProperty<?>` implementations, but the means to do that are not covered in this article. The vanilla code provides several convenience implementations:
 
-It follows very easily from this that since basic `BlockState`'s are generated into a fixed set at startup, you are able and encouraged to use reference comparison (==) to check if they are equal!
+  * `IntegerProperty`
+    * Implements `IProperty<Integer>`. Defines a property that holds an integer value.
+    * Created by calling `IntegerProperty.create(String propertyName, int minimum, int maximum)`.
+  * `BooleanProperty`
+    * Implements `IProperty<Boolean>`. Defines a property that holds a `true` or `false` value.
+    * Created by calling `BooleanProperty.create(String propertyName)`.
+  * `EnumProperty<E extends Enum<E>>`
+    * Implements `IProperty<E>`. Defines a property that can take on the values of an Enum class.
+    * Created by calling `EnumProperty.create(String propertyName, Class<E> enumClass)`.
+    * It is also possible to use only a subset of the Enum values (e.g. 4 out of 16 `DyeColor`s). See the overloads of `EnumProperty.create`.
+  * `DirectionProperty`
+    * This is a convenience implementation of `EnumProperty<Direction>`
+    * Several convenience predicates are also provided. For example, to get a property that represents the cardinal directions, call `DirectionProperty.create("<name>", Direction.Plane.HORIZONTAL)`; to get the X directions, `DirectionProperty.create("<name>", Direction.Axis.X)`
 
+Different blocks may share the same `IProperty<?>` object. Vanilla generally has separate ones for every single block, but it is merely personal preference.
```

Comment:
Should probably mention `BlockStateProperties` and to *avoid* making copies of vanilla properties.

Reply:
Changed to mention `BlockStateProperties`, and reworded some phrases to suggest referencing instead of only creating.